### PR TITLE
Don't allow leading or trailing spaces in the level name

### PR DESCRIPTION
**Pull request by joshlory**
I don't know why anyone would do this, but there were 4 levels that differed only by a space at the end of the level name.  This causes problems because looking up a level by name in Rails will ignore any trailing whitespace:

> All MySQL collations are of type PADSPACE. This means that all CHAR, VARCHAR, and TEXT values in MySQL are compared without regard to any trailing spaces. “Comparison” in this context does not include the LIKE pattern-matching operator, for which trailing spaces are significant.

Added a check, a test, and deleted the offending levels (all levels were identical to the version without the space, except for `2-3 Maze sq1 .level` which I renamed to `2-3 Maze sq1.level`).

### CODE REVIEW

File: dashboard/app/controllers/levels_controller.rb [hunk 1]
```diff
@@ -79,13 +79,18 @@ def update_blocks
...
-      @level.errors.add(:name, 'Cannot change only the capitalization of the level name (it confuses git on OSX)')
-      render json: @level.errors, status: :unprocessable_entity
-      return
+    if level_params[:name] && @level.name != level_params[:name]
+      if @level.name.downcase == level_params[:name].downcase
+        # do not allow case-only changes in the level name because that confuses git on OSX
+        @level.errors.add(:name, 'Cannot change only the capitalization of the level name (it confuses git on OSX)')
+        render json: @level.errors, status: :unprocessable_entity
+        return
+      elsif @level.name == level_params[:name].strip
```
Comment (laurelfan) [inline dashboard/app/controllers/levels_controller.rb]: hm, should we just always strip the name param? I can't imagine anyone is doing this intentionally

Comment (joshlory) [inline dashboard/app/controllers/levels_controller.rb]: Fixed in 2d10240b313e320098c65a04073826f1e908e4cf.

File: dashboard/app/controllers/levels_controller.rb [hunk 2]
```diff
@@ -79,6 +79,7 @@ def update_blocks
   # PATCH/PUT /levels/1
   # PATCH/PUT /levels/1.json
   def update
+    level_params[:name] = level_params[:name].strip
```
Comment (laurelfan) [inline dashboard/app/controllers/levels_controller.rb]: should we also do it on create? might be easier to do it in the Level model as a before_validation callback, eg:
http://stackoverflow.com/questions/3268053/rails-before-validation-strip-whitespace-best-practices

**Comment by laurelfan** [comment]
lgtm!

**Pull request merged**